sdk/services/kwentaToken.ts

```diff
@@ -0,0 +1,469 @@
+		const KwentaTokenContract = new EthCallContract(KwentaToken.address, ERC20ABI);
+		const SupplyScheduleContract = new EthCallContract(SupplySchedule.address, SupplyScheduleABI);
+		const vKwentaTokenContract = new EthCallContract(vKwentaToken.address, ERC20ABI);
+		const MultipleMerkleDistributorContract = new EthCallContract(
+			MultipleMerkleDistributor.address,
+			MultipleMerkleDistributorABI
+		);
+		const veKwentaTokenContract = new EthCallContract(veKwentaToken.address, ERC20ABI);
+
+		const [
+			rewardEscrowBalance,
+			stakedNonEscrowedBalance,
+			stakedEscrowedBalance,
+			claimableBalance,
+			kwentaBalance,
+			weekCounter,
+			totalStakedBalance,
+			vKwentaBalance,
+			vKwentaAllowance,
+			kwentaAllowance,
+			epochPeriod,
+			veKwentaBalance,
+			veKwentaAllowance,
+		]: ethers.BigNumber[] = await this.sdk.context.multicallProvider.all([
+			RewardEscrowContract.balanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.nonEscrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.escrowedBalanceOf(this.sdk.context.walletAddress),
+			KwentaStakingRewardsContract.earned(this.sdk.context.walletAddress),
+			KwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			SupplyScheduleContract.weekCounter(),
+			KwentaStakingRewardsContract.totalSupply(),
+			vKwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			vKwentaTokenContract.allowance(this.sdk.context.walletAddress, vKwentaRedeemer.address),
+			KwentaTokenContract.allowance(this.sdk.context.walletAddress, KwentaStakingRewards.address),
+			MultipleMerkleDistributorContract.distributionEpoch(),
+			veKwentaTokenContract.balanceOf(this.sdk.context.walletAddress),
+			veKwentaTokenContract.allowance(this.sdk.context.walletAddress, veKwentaRedeemer.address),
+		]);
+
+		return {
+			rewardEscrowBalance: wei(rewardEscrowBalance),
+			stakedNonEscrowedBalance: wei(stakedNonEscrowedBalance),
+			stakedEscrowedBalance: wei(stakedEscrowedBalance),
+			claimableBalance: wei(claimableBalance),
+			kwentaBalance: wei(kwentaBalance),
+			weekCounter: Number(weekCounter),
+			totalStakedBalance: wei(totalStakedBalance),
+			vKwentaBalance: wei(vKwentaBalance),
+			vKwentaAllowance: wei(vKwentaAllowance),
+			kwentaAllowance: wei(kwentaAllowance),
+			epochPeriod: Number(epochPeriod),
+			veKwentaBalance: wei(veKwentaBalance),
+			veKwentaAllowance: wei(veKwentaAllowance),
+		};
+	}
+
+	public async getEscrowData() {
+		const { RewardEscrow } = this.sdk.context.contracts;
+
+		if (!RewardEscrow) {
+			throw new Error(sdkErrors.UNSUPPORTED_NETWORK);
+		}
+
+		const RewardEscrowContract = new EthCallContract(RewardEscrow.address, RewardEscrowABI);
+
+		const schedules = await RewardEscrow.getVestingSchedules(
+			this.sdk.context.walletAddress,
+			0,
+			1000
+		);
+
+		const vestingSchedules = schedules.filter((schedule) => schedule.escrowAmount.gt(0));
+
+		const calls = [];
+
+		for (const schedule of vestingSchedules) {
```

Comment:
I think we talked about using this for awaiting promises inside but if we're just pushing calls we can go with a simpler `.map`

Reply:
Yeah, I know. I personally prefer using for loops outside components because they're quite a bit faster, and we're not really working in a FP context here. It might be something interesting to discuss though, since we should probably be setting standard for these things anyway. I just figured we could take a perf win wherever we can find one.

Reply:
I think if we were iterating over thousands of items or was a specific area we were looking to optimise it might be something to consider but where there would really be no noticeable gain I think we should always go with the cleaner code option and remain consistent with our other iteration patterns throughout the codebase.